### PR TITLE
Convert reddit links to a valid video link

### LINKED ISSUE #23: [youtube-dl] reddit posts somtimes dont work, when pointing at imgur links

### LINKED ISSUE #24: [youtube-dl] multi-image-posts don't work

### PR DESCRIPTION
**Pull request by Zeboto**

### CODE REVIEW

File: alexBot/cogs/video_dl.py
Comment (xela-zone) [inline alexBot/cogs/video_dl.py]: should probably use aiohttp instead of requests

see https://github.com/mralext20/alex-bot/blob/main/alexBot/cogs/sugery.py#L52 for example of using the get_json helper from ..tools

**Review by xela-zone — CHANGES_REQUESTED** [review]

File: alexBot/cogs/video_dl.py
Comment (7bitlyrus) [inline alexBot/cogs/video_dl.py]: Revert change for subdomains (`w{0,3}`), original regex was designed to match some more archaic Reddit subdomain schemes, e.g.:

https://old.reddit.com/r/aww/comments/90bu6w
https://aww.reddit.com/comments/90bu6w

File: alexBot/cogs/video_dl.py
Comment (7bitlyrus) [inline alexBot/cogs/video_dl.py]: This approach (hardcorded index) isn't incredibly maintainable. A better approach is to have a separate list of regexs and check if in the seperate regex list

**Review by 7bitlyrus — CHANGES_REQUESTED** [review]

**Review by xela-zone — APPROVED** [review]

**Comment by xela-zone** [comment]
i've tested this, it works

**Comment by 7bitlyrus** [comment]
> It is not possible to use aiohttp instead of httpx for some reason. I have tested extensively with requests, aiohttp, and httpx. requests and httpx will always work, while aiohttp will always return too many requests.
\- @Zeboto 


Is it not possible this is a the result of aiohttp's user-agent? Did you test that?

**Review by 7bitlyrus — COMMENTED** [review]
Everything looks good. Just waiting for a response on whether aiohttp's UA is the cause of 429s before approval.

**Comment by 7bitlyrus** [comment]
> Everything looks good. Just waiting for a response on whether aiohttp's UA is the cause of 429s before approval.

Yeah, I looked into this. No clue why 429s are resulting, but it keeps to be due to aiohttp capitalizing a header or some SSL difference.

**Review by 7bitlyrus — APPROVED** [review]

**Comment by 7bitlyrus** [comment]
After looking into it a bit more, I found a similar issue with aiohttp and the Telsa API returning 403s: https://github.com/aio-libs/aiohttp/issues/5643.

Drawing from the discussion on that issue, it looks like Reddit is returning a 429 for SSL issues. This code to override the SSL ciphers resolves the aiohttp 429 issue for me (where setting the UA alone did not):
```py
import aiohttp, asyncio, ssl

async def test():
    async with aiohttp.ClientSession() as session:
        headers = {'user-agent': 'AlexBot:v1.0.0'}
        url = 'https://www.reddit.com/r/funny/comments/5gn8ru.json'
        
        FORCED_CIPHERS = (
            'ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+HIGH:'
            'DH+HIGH:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+HIGH:RSA+3DES'
        )
        sslcontext = ssl.create_default_context()
        sslcontext.set_ciphers(FORCED_CIPHERS)

        resp = await session.get(url=url, headers=headers, ssl=sslcontext)
        print(resp)

loop = asyncio.get_event_loop()
loop.run_until_complete(test())
```

**Pull request merged**
